File: 0. Project 심다/simda/src/main/java/ssafy/a709/simda/dto/FollowDto.java
package ssafy.a709.simda.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ssafy.a709.simda.domain.Follow;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class FollowDto {

    // FollowId
    private int followId;
    // Many(From UserId) to One(UserId)
    private UserDto fromUserId;
    // Many(To UserId) to One(UserId)
    private UserDto toUserId;

    public static FollowDto changeToFollowDto(Follow follow) {

        return FollowDto.builder()
                .followId(follow.getFollowId())
                .fromUserId(UserDto.changeToUserDto(follow.getFromUserId()))
                .toUserId(UserDto.changeToUserDto(follow.getToUserId()))
                .build();
    }

}

File: frontend/simda/lib/models/FollowDto.dart
import 'UserDto.dart';

class FollowDto {
  final int ? followId;
  final UserDto fromUserId;
  final UserDto toUserId;

  FollowDto({
    this.followId,
    required this.fromUserId,
    required this.toUserId,
  });

  Map<String, dynamic> toJson() {
    return {
      'followId': followId,
      'fromUserId': fromUserId,
      'toUserId': toUserId,
    };
  }

  // factory FollowDto.fromJson(Map<String, dynamic> json) {
  //   return FollowDto(
  //       followId: json['followId'],
  //       fromUserId: json['fromUserId'],
  //       toUserId: json['toUserId'],
  //   );
  // }
}